app/reschedule/ensure-scheduled.js

```diff
@@ -0,0 +1,14 @@
+const createSchedule = require('../inbound/create-schedule')
+const abandonSchedule = require('./abandon-schedule')
+const getExistingSchedule = require('./get-existing-schedule')
+
+const ensureScheduled = async (paymentRequestId, schemeId, transaction) => {
+  const existingSchedule = await getExistingSchedule(paymentRequestId, transaction)
+  if (!existingSchedule) {
+    await createSchedule(schemeId, paymentRequestId, transaction)
+  } else {
+    await abandonSchedule(existingSchedule.scheduleId, transaction)
+  }
+}
```

Comment:
Tests for this are needed

Reply:
Tests already exist from when this was originally created -> https://github.com/DEFRA/ffc-pay-processing/blob/17598d11e4c3ad52b380053f8908de6a27087f88/test/integration/local/reschedule.test.js#L47 and https://github.com/DEFRA/ffc-pay-processing/blob/17598d11e4c3ad52b380053f8908de6a27087f88/test/integration/local/reschedule.test.js#L53

Do you think this is sufficient?

Reply:
I think we should have unit tests for the functions that make up these integration tests, i.e. `getExistingHold`, `addHold` and `ensureScheduled`

Reply:
I'm not so sure we do.  If we've already got higher level tests that cover all behaviour, then repeating sub steps in lower level unit tests add no value.

I've looked through all the higher level tests in narrow and local and looks like every possible scenario is covered already.

If you really feel this warrants lower level tests I'm happy to trust your judgement and will add them.

Reply:
Just looking through I csan understand your point, if we test each of these functions in-term we are essentially testing that the Sequilze module as 99% of them are reading/writing/updating the DB -- happy to omit these and retract my suggestion